runtime/lua/vim/lsp/document_color.lua

```diff
@@ -15,9 +15,10 @@ local M = {}
 
 --- @class (private) vim.lsp.document_color.BufState
 --- @field enabled boolean Whether document_color is enabled for the current buffer
---- @field buf_version? integer Buffer version for which the color ranges correspond to
+--- @field processed_version table<integer, integer?> (client_id -> buffer version) Buffer version for which the color ranges correspond to
 --- @field applied_version table<integer, integer?> (client_id -> buffer version) Last buffer version for which we applied color ranges
 --- @field hl_info table<integer, vim.lsp.document_color.HighlightInfo[]?> (client_id -> color highlights) Processed highlight information
+--- @field ns table<integer, integer?> (client_id -> namespace ID) documentColor namespace ID for each client.
```

Comment:
A minor nitpick: storing `client_id -> namespace ID` map for every buffer client attaches to (although it doesn't depend on the buffer) seems like not the best abstraction. It is not wrong per se to store duplicating info and memory overhead is not huge (although scales `O(nm)` instead of `O(n)`), but still.

Perhaps an extra `clientstates` for (at the moment) only a client's namespace id is a better long-term approach here?

Reply:
I don’t fully understand the suggestion. Are you saying I could use a single namespace?

Reply:
No, on the contrary. Having separate namespaces for each client is good. And that is indeed what is done in the PR (`api.nvim_create_namespace('nvim.lsp.document_color.client_' .. client_id)`).

But as namespace depends only on the client id, storing namespace identifiers inside state for each buffer (`bufstates[bufnr].ns[client_id]`) leads to duplicating information: `bufstates[1].ns[1]` is the same as `bufstates[2].ns[1]`, etc. That in itself is not a huge issue (yet), but leads to wrong abstraction.

Storing namespace identifiers in a more direct map, so it can be accessed as `client_ns[client_id]` (or `clientstates[client_id].ns`), is more future-proof.

Reply:
Ohhhh gotcha, good point. I’ll fix it!